libsnowflakeclient/lib/error.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include <string.h>
+#include "error.h"
+#include "snowflake_memory.h"
+
+void *STDCALL set_snowflake_error(SNOWFLAKE_ERROR *error,
+                                  SNOWFLAKE_ERROR_CODE error_code,
+                                  const char *msg,
+                                  const char *sfqid,
+                                  const char *file,
+                                  int line) {
+    size_t sfqid_size = 1;
+    // NULL error passed in. Should never happen
+    if (!error) {
+        return NULL;
+    }
+
+    // Allocate enough memory for sfqid
+    sfqid_size += strlen(sfqid);
+    error->sfqid = (char *) SF_CALLOC(1, sfqid_size);
```

Comment:
probably we need to revisit this. If sfqid is given, you don't need to copy here but pass the pointer to sfqid. This way, you won't need to free memory for sfqid.

Reply:
yeah we have to be careful with memory allocation in C so it doesn't leak.

Reply:
I've changed this to be a fixed size char array.